gui/pregnancy.lua

```diff
@@ -0,0 +1,328 @@
+        for index, relation in pairs(historical_unit.histfig_links) do
+            if relation._type == df.histfig_hf_link_spousest then
+                historical_spouse=df.historical_figure.find(relation.target_hf)
+                if not historical_spouse then culled = true --there was an id, but there wasn't a histfig with that id (due culling)
+                elseif df.global.plotinfo.site_id==historical_spouse.info.whereabouts.site then--i dont think this should ever be true
+                    spouse_loc = 'local'
+                else spouse_loc = 'offsite'
+                end
+            end
+        end
+    end
+    if culled then 
+        return {'Unknown','culled'}
+    else 
+        return {dfhack.TranslateName(historical_spouse.name),spouse_loc,historical_spouse}
+    end
+end
+
+function PregnancyGui:CreatePregnancy()
+	local genes,father_id,father_caste,father_name
+    local bypass = true
+    local force = self.subviews.Force:getOptionValue()
+
+    local count = #self.msg
+    for i=0, count do self.msg[i]=nil end --empty self.msg
+
+    if self.father then
+        genes=self.father.appearance.genes:new()
+        father_id=self.father.hist_figure_id
+        father_caste=self.father.caste
+        father_name=self:findName(self.father)
+    else
+        genes=self.mother.appearance.genes:new()--i dont think historical figures have genes
+        father_id=self.father_historical.id
+        father_caste=self.father_historical.caste
+        father_name=self:findName(self.father_historical)
+    end
+
+    if self.mother.pregnancy_timer > 0 then
+        local og_father = df.historical_figure.find(self.mother.pregnancy_spouse)
+        bypass = false
+        if force and og_father then 
+            table.insert(self.msg, ('SUCCESS:%sMother:%s%sFather:%s%sPrevious pregnancy with %s aborted'):format(
+            NEWLINE,    
+            self:findName(self.mother),
+            NEWLINE, 
+            father_name,
+            NEWLINE,
+            dfhack.TranslateName(og_father.name)
+            ))
+        elseif force then 
+            table.insert(self.msg, ('SUCCESS:%sMother:%s%sFather:%s%sPrevious pregnancy aborted'):format(
+            NEWLINE,    
+            self:findName(self.mother),
+            NEWLINE, 
+            father_name,
+            NEWLINE
+            ))
+        elseif og_father then
+            table.insert(self.msg, ('FAILED:%s%s already pregnant with %s%s'):format(
+                NEWLINE,
+                self:findName(self.mother),
+                dfhack.TranslateName(og_father.name),
+                force
+            ))
+        else 
+            table.insert(self.msg, ('FAILED:%s%s is already pregnant, no father is recorded'):format(
+                NEWLINE,
+                self:findName(self.mother)
+        ))
+        end
+    end
+    -- self.success = false
+    if bypass or force then
+        --TODO add GUI to select the number of months for pregnancy timer
```

Comment:
You can use a RangeSlider for this. See example here: https://github.com/DFHack/dfhack/blob/develop/plugins/lua/zone.lua#L169-L217

Reply:
Thanks! I got it working (kinda) i had to reverse index my values and use the index for the get_left_idx_fn and get_right_idx_fn.

Reply:
Re the RangeSlider - I have some issues dragging the slider to the minimum value (for me I have to slide all the way to the right to select the minimum).
Its not the biggest deal.

I did have to make the window pretty wide (initially i was planning to give the options in weeks). I guess RangeSlider is most suited for when there are <10 or so options?

I am curious if the reverse indexing was correct too.